c/cmake_modules/GoUtils.cmake

```diff
@@ -180,36 +201,79 @@ function(add_go_lib GO_MOD_DIR GO_LIBNAME)
       list(APPEND GO_ENV_VARS "GOARCH=arm64")
     endif()
 
-    add_custom_command(OUTPUT "${LIBOUT_SHARED}.${ADBC_FULL_SO_VERSION}"
-                       WORKING_DIRECTORY ${GO_MOD_DIR}
-                       DEPENDS ${ARG_SOURCES}
-                       COMMAND ${CMAKE_COMMAND} -E env ${GO_ENV_VARS} ${GO_BIN} build
-                               ${GO_BUILD_TAGS} "${GO_BUILD_FLAGS}" -o
-                               ${LIBOUT_SHARED}.${ADBC_FULL_SO_VERSION}
-                               -buildmode=c-shared ${GO_LDFLAGS} .
-                       COMMAND ${CMAKE_COMMAND} -E remove -f
-                               "${LIBOUT_SHARED}.${ADBC_SO_VERSION}.0.h"
-                       COMMENT "Building Go Shared lib ${GO_LIBNAME}"
-                       COMMAND_EXPAND_LISTS)
+    if(WIN32)
```

Comment:
We're now fully duplicating the build paths...if Windows doesn't want the version suffix, can it just rename the file afterwards instead?

Reply:
Yes ideally the Windows files don't need the version in the file name, as that makes copying them during build events in Visual Studio difficult. Where is the duplication happening? The main change was to remove `${ADBC_FULL_SO_VERSION}` and `${ADBC_SO_VERSION}` suffixes.

Reply:
The build commands are effectively the same, though. If it's just a matter of filename, can we either (1) introduce a variable to hold the filename, or (2) just rename it after build?